Gale Databases.js

```diff
@@ -25,93 +26,222 @@
 	GNU Affero General Public License for more details.
 
 	You should have received a copy of the GNU Affero General Public License
-	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
 
 	***** END LICENSE BLOCK *****
 */
 
 
-function processMultipleEntries(entries) {
-	var keyValuePairs = {};
-	for (var entry of entries) {
-		keyValuePairs[entry.href] = entry.textContent;
+function detectWeb(doc, url) {
+	if (url.includes('/ps/eToc.do')) {
+		return "book";
 	}
-	Zotero.selectItems(keyValuePairs, function (selectedItems) {
-		if (selectedItems) {
-			Zotero.Utilities.processDocuments(Object.keys(selectedItems), processSingleEntry);
-		}
-	});
-}
-
-function processSingleEntry(doc) {
-	var entry = doc.querySelector('.zotero');
-	var docId = entry.getAttribute('data-documentnumber');
-	var documentUrl = entry.getAttribute('href');
-	var productName = entry.getAttribute('data-productname');
-	var documentData = '{"docId":"' + docId + '","documentUrl":"' + documentUrl + '","productName":"' + productName + '"}';
-	var urlParams = "citationFormat=RIS&documentData=" + encodeURIComponent(documentData).replace(/%20/g, "+");
-	Zotero.Utilities.doPost("/ps/citationtools/rest/cite/download", urlParams, translate);
-}
-
-function translate(data) {
-	var translator = Zotero.loadTranslator("import");
-	// use RIS translator
-	translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
-	translator.setString(transform(data));
-	translator.setHandler("itemDone", function (obj, item) {
-		if (item.ISSN) {
-			item.ISSN = Zotero.Utilities.cleanISSN(item.ISSN);
-		}
-		if (item.pages && item.pages.endsWith("+")) {
-			item.pages = item.pages.replace(/\+/, "-");
-		}
-		item.attachments.push({ document: data, title: "Snapshot" });
-		item.complete();
-	});
-	translator.translate();
-}
-
-function transform(ris) {
-	return ris.trim()
-		.replace(/M1\s*-/g, "IS  -") // gale puts issue numbers in M1
-		.replace(/^(?:L2|M2)\s+-.+\n/gm, '') // Ignore
-		.replace(/^SP\s+-\s+NA\n/gm, '') // Remove missing page numbers
-		.replace(/^N1(?=\s+-\s+copyright)/igm, 'CR');
-}
-
-function getCitableDocuments(doc) {
-	return doc.getElementsByClassName('zotero');
+	else if (url.includes('Search.do') && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	else if (doc.querySelector('a[data-gtm-feature="bookView"]')) {
+		return "bookSection";
+	}
+	return "magazineArticle";
 }
 
-
-function detectWeb(doc, _url) {
-	if (doc.getElementById('searchResults')) {
-		Zotero.monitorDOMChanges(doc.querySelector('#searchResults'));
-	}
-	var entries = getCitableDocuments(doc);
-	switch (entries.length) {
-		case 0: return false;
-		case 1: return entries[0].getAttribute('data-zoterolabel');
-		default: return 'multiple';
+function getSearchResults(doc, checkOnly) {
+	let items = {};
+	let found = false;
+	let rows = doc.querySelectorAll('h3.title>a.documentLink');
```

Comment:
Add whitespace around `>` (and we should fix the Scaffold template if it does that)